src/insights_mcp/client.py

```diff
@@ -291,8 +286,75 @@ async def make_request(self, fn, *args, **kwargs) -> dict[str, Any] | str:
                     await self.fetch_token()
             except OAuthError as e:
                 raise ValueError(self.no_auth_error(e)) from e
+
+    async def make_request(self, fn, *args, **kwargs) -> dict[str, Any] | str:
+        """Make an HTTP request with OAuth2 token management.
+
+        Handles token refresh when needed and supports OAuth middleware.
+
+        Args:
+            fn: HTTP method function to call
+            *args: Positional arguments for the HTTP method
+            **kwargs: Keyword arguments for the HTTP method
+
+        Returns:
+            JSON response data or error information
+        """
+        if not self.oauth_enabled and self.refresh_token is None and self.client_secret is None:
+            return self.no_auth_error(ValueError("Client not authenticated"))
+
+        await self.refresh_auth()
+
         return await super().make_request(fn, *args, **kwargs)
 
+    async def decode_token(self) -> dict[str, Any] | None:
+        """Decode the JWT access token and return its payload.
+
+        Note: authlib's OAuth2Token does not provide JWT decoding capabilities.
+        While authlib.jose.jwt exists, it requires signature verification which
+        is not needed here since we're just reading claims. PyJWT is used instead
+        as it supports decoding without verification and is already a dependency.
+
+        Returns:
+            Decoded token payload as a dictionary, or None if token is not available or invalid.
+        """
+        await self.refresh_auth()
+        if not self.token or "access_token" not in self.token:
+            return None
+        try:
+            # Decode without verification (since we're just reading claims, not validating)
+            # In production, you might want to verify the signature
+            decoded = jwt.decode(
+                self.token["access_token"],
+                options={"verify_signature": False},
+                algorithms=["RS256"],
+            )
+            return decoded
```

Comment:
Given this `decode_token()` could be called separately in the `get_org_id()` and `get_user_id()` below, what about caching the result `decoded` ? say, it could like: 

```
        ...
        if self._decoded_token:
            return self._decoded_token
        try:
            self._decoded_token = jwt.decode( ... )
            ....          
```

Reply:
I'm not really concerned about run-time here - but it's nicer for sure!